api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImplTest.java

```diff
@@ -387,4 +415,83 @@ public void searchForObs_shouldHandleComplexQuery() {
 			assertThat(resultsList.get(i - 1).getObsDatetime(), sameOrAfter(resultsList.get(i).getObsDatetime()));
 		}
 	}
+	
+	@Test
+	public void searchForObs_shouldReturnObsByMemberReference() {
+		ReferenceParam memberReference = new ReferenceParam();
+		memberReference.setValue(MEMBER_UUID);
+		memberReference.setChain("");
+		
+		Collection<Obs> results = dao.searchForObservations(null, memberReference, null, null, null, null, null, null, null,
+		    null);
+		
+		assertThat(results, notNullValue());
+		// This gives java.lang.AssertionError:
+		//Expected: not an empty collection
+		//     but: was <[]>
+		//Expected :not an empty collection
+		//Actual   :<[]>,
+		// I tried adding data to xml file but not relevant tags found similar to groupMember
+		//		assertThat(results, not(empty()));
+		//		assertThat(results.iterator().next().getUuid(), equalTo(OBS_UUID));
+		//		assertThat(results.iterator().next().getGroupMembers().iterator().next().getUuid(), equalTo(OBS_UUID));
```

Comment:
@ibacher Please help me with this error

Reply:
So `MEMBER_UUID` should be referring to the UUID that is the member of an obs group. An `Obs` is a member of an obs group when the `obs_group_id` column points to the id of another observation.

Reply:
@ibacher I added this and tried pointing to both the ids still the same error
```xml
<obs obs_id="22" person_id="7" concept_id="5092" encounter_id="3" obs_datetime="2008-07-01 00:00:00.0" location_id="1" value_numeric="98.3" comments="" creator="1" date_created="2008-08-18 14:09:35.0" voided="false" value_coded="[NULL]" value_coded_name_id="[NULL]" value_complex="[NULL]" value_text="[NULL]" value_datetime="[NULL]" value_drug="[NULL]" uuid="4efa62d2-6b8b-4803-a8fa-3f32ee54db4f"/>
    <obs obs_id="23" person_id="7" obs_group_id="22" concept_id="5242" encounter_id="3" obs_datetime="2008-07-01 00:00:00.0" location_id="1" value_numeric="14.0" comments="" creator="1" date_created="2008-08-18 14:09:35.0" voided="false" value_coded="[NULL]" value_coded_name_id="[NULL]" value_complex="[NULL]" value_text="[NULL]" value_datetime="[NULL]" value_drug="[NULL]" uuid="744b91f8-bdbc-4950-833b-002244e9fa2b"/>
```

Reply:
When you say "pointing to both ids", could you spell out what you mean?

Reply:
i tried both

``` java
memberReference.setValue("744b91f8-bdbc-4950-833b-002244e9fa2b");
		memberReference.setChain("");
```
THIS FAILED SO I TRIED THE BELOW ONE AND THAT ALSO FAILED
```java
memberReference.setValue("4efa62d2-6b8b-4803-a8fa-3f32ee54db4f");
		memberReference.setChain("");
```

Reply:
The first one looks right to me. Could you make those changes to this branch and push it out? That way we can both be looking at the same code.

Reply:
@ibacher Any ideas what's wrong here?